### PR TITLE
linting tests requiring nf-test will be warnings instead of failures

### PR DESCRIPTION
**Pull request by mirpedrol**
## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

### CODE REVIEW

File: nf_core/modules/lint/module_tests.py [hunk 1]
```diff
@@ -104,7 +104,7 @@ def module_tests(_, module):
             with open(pytest_yml_path, "r") as fh:
                 pytest_yml = yaml.safe_load(fh)
                 if module.component_name in pytest_yml.keys():
-                    module.failed.append(
```
Comment (mashehu) [inline nf_core/modules/lint/module_tests.py:107]: This one can stay as a warning

File: nf_core/subworkflows/lint/subworkflow_tests.py [hunk 2]
```diff
@@ -114,7 +114,7 @@ def subworkflow_tests(_, subworkflow):
             with open(pytest_yml_path, "r") as fh:
                 pytest_yml = yaml.safe_load(fh)
                 if "subworkflows/" + subworkflow.component_name in pytest_yml.keys():
-                    subworkflow.failed.append(
```
Comment (mashehu) [inline nf_core/subworkflows/lint/subworkflow_tests.py:117]: This one can stay as well

**Review by mashehu — APPROVED** [review]

**Review by asp8200 — APPROVED** [review]
I need this

**Review by pinin4fjords — APPROVED** [review]
Yes please!

**Pull request merged**